From ba7c4da06b6b1af965850aac33d63814ff8b290f Mon Sep 17 00:00:00 2001
From: SeongJae Park <sj@kernel.org>
Date: Fri, 6 Jan 2023 17:18:57 +0000
Subject: [PATCH 45/45] include/linux/mm: declare different type of split_vma()
 for !CONFIG_MMU

Commit 95000286ff79 ("mm: switch vma_merge(), split_vma(), and
__split_vma to vma iterator") on mm-unstable didn't make the change to
nommu.c, but the declaration change was applied to !CONFIG_MMU, too.  It
causes below build failure.

    linux/mm/nommu.c:1347:5: error: conflicting types for 'split_vma'
     1347 | int split_vma(struct mm_struct *mm, struct vm_area_struct *vma,
          |     ^~~~~~~~~
    In file included from linux/mm/nommu.c:20:
    linux/include/linux/mm.h:2846:12: note: previous declaration of 'split_vma' was here
     2846 | extern int split_vma(struct vma_iterator *vmi, struct vm_area_struct *,
          |            ^~~~~~~~~

Fix the build failure by adding the split_vma() declaration for
!CONFIG_MMU case.

Fixes: 95000286ff79 ("mm: switch vma_merge(), split_vma(), and __split_vma to vma iterator") on mm-unstabl
Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/mm.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/linux/mm.h b/include/linux/mm.h
index 47298a56c..cdc7b6976 100644
--- a/include/linux/mm.h
+++ b/include/linux/mm.h
@@ -2726,8 +2726,13 @@ extern struct vm_area_struct *vma_merge(struct vma_iterator *vmi,
 extern struct anon_vma *find_mergeable_anon_vma(struct vm_area_struct *);
 extern int __split_vma(struct vma_iterator *vmi, struct vm_area_struct *,
 		       unsigned long addr, int new_below);
+#ifdef CONFIG_MMU
 extern int split_vma(struct vma_iterator *vmi, struct vm_area_struct *,
 			 unsigned long addr, int new_below);
+#else
+extern int split_vma(struct mm_struct *, struct vm_area_struct *,
+			 unsigned long addr, int new_below);
+#endif	/* CONFIG_MMU */
 extern int insert_vm_struct(struct mm_struct *, struct vm_area_struct *);
 extern void unlink_file_vma(struct vm_area_struct *);
 extern struct vm_area_struct *copy_vma(struct vm_area_struct **,
-- 
2.39.0.rc2.1.gbd5df96b79

